Lesson 4/Lesson 4/Lesson 4.cs

```diff
@@ -0,0 +1,236 @@
+    // Then, copy elements that are not equal to 0 to arrayWithoutZeros.
+    // And return array with odd numbers only without zeros.
+    int[] arrayWithoutZeros = new int[numOfOddNumbers];
+    for (int element = 0; element < oddNumbers.Length; element++)
+    {
+        if (oddNumbers[element] != 0)
+        {
+            arrayWithoutZeros[elementsWithoutZeros] = oddNumbers[element];
+            elementsWithoutZeros++;
+        }
+    }
+
+    return arrayWithoutZeros;
+}
+
+string[] ReplaceNumbersWithLetters(int[] array)
+{
+    // Replace numbers from the array with letters of the English alphabet using ASCII and
+    // add these letters in the englishLetters array.
+    // If the number in array equal to 1 it will be replaced by 'a',
+    // if it's 2 it will be replaced by 'b' and so on
+    string[] englishLetters = new string[array.Length];
+    for (int element = 0; element < array.Length; element++)
+    {
+        char englishLetter = (char)('a' + array[element] - 1);
+        englishLetters[element] = englishLetter.ToString();
+    }
+
+    return englishLetters;
+}
+
+void DisplayLettersArray(string[] array)
+{
+    foreach (string element in array)
+    {
+        Console.Write($"{element} ");
+    }
+}
+
+string[] ChangeLettersToUpperCase(string[] array)
+{
+    // It letters "a","e","i","d","h","j" are present in array they will become uppercase letters.
+    // If it's another letters they will remain lowercase.
+    string[] uppercaseLetters = new string[array.Length];
+    for (int element = 0; element < array.Length; element++)
+    {
+        if (array[element] == "a" || array[element] == "e" || array[element] == "i"
+            || array[element] == "d" || array[element] == "h" || array[element] == "j")
+        {
+            string upper = array[element].ToUpper();
+            uppercaseLetters[element] = upper;
+        }
+        else
+        {
+            uppercaseLetters[element] = array[element];
+        }
+    }
+
+    return uppercaseLetters;
+}
+
+int CheckHowManyUppercaseLetters(string[] array)
+{
+    // Check how many uppercase letters in the array and return number of those letters
+    int count = 0;
+    foreach (string element in array)
+    {
+        if (element.All(char.IsUpper))
+        {
+            count++;
+        }
+    }
+
+    return count;
+}
+
+void CheckBiggerNumber(int firstNumber, int secondNumber)
+{
+    Console.Write($"First array contains {firstNumber} uppercase letters, " +
+            $"Second array contains {secondNumber} uppercase letters. ");
+    if (firstNumber > secondNumber)
+    {
+        Console.Write($"First array contains more uppercase letters.");
+    }
+    else if (firstNumber < secondNumber)
+    {
+        Console.Write($"Second array contains more uppercase letters.");
+    }
+    else if (firstNumber == 0 && secondNumber == 0)
+    {
+        Console.Write($"Arrays don't contain uppercase letters.");
+    }
```

Comment:
Is this case possible?

Reply:
​Yes, this is possible if users don't have letters a, e, i, d, h, j in arrays. If he entered a small number into the input at the beginning